src/components/questionTypes/SC/SCEditQuestionOption.js

```diff
@@ -0,0 +1,96 @@
+import React from 'react'
+import PropTypes from 'prop-types'
+import classNames from 'classnames'
+import { compose, withHandlers, withState } from 'recompose'
+import { Icon } from 'semantic-ui-react'
+import styles from './styles'
+
+const propTypes = {
+  name: PropTypes.string.isRequired,
+  editMode: PropTypes.bool.isRequired,
+  toggleEditMode: PropTypes.func.isRequired,
+  saveNewName: PropTypes.func.isRequired,
+  discardNewName: PropTypes.func.isRequired,
+  handleNameChange: PropTypes.func.isRequired,
+  handleKeyPress: PropTypes.func.isRequired,
+}
+
+const SCEditQuestionOption = ({
+  name,
+  editMode,
+  toggleEditMode,
+  saveNewName,
+  discardNewName,
+  handleNameChange,
+  handleKeyPress,
+}) => (
+  <div className="optionEditCont">
+    <div className="optionEdit">
+      {editMode ? (
+        <input type="text" value={name} onChange={handleNameChange} onKeyDown={handleKeyPress} />
+      ) : (
+        <div className="name">{name}</div>
+      )}
+    </div>
+
+    <div className="editOptBtnRight">
+      {editMode ? (
+        <button className={classNames('rightAction', { editMode })} type="button" onClick={saveNewName}>
+          <Icon name="save" />
+        </button>
+      ) : null}
+      {editMode ? (
+        <button className={classNames('rightAction', { editMode })} type="button" onClick={discardNewName}>
+          <Icon name="delete" />
+        </button>
+      ) : null}
+      {!editMode ? (
+        <button className={classNames('rightAction', { editMode })} type="button" onClick={toggleEditMode}>
+          <Icon name="edit" />
+        </button>
+      ) : null}
+    </div>
+    <style jsx>{styles}</style>
+  </div>
+)
+
+SCEditQuestionOption.propTypes = propTypes
+
+export default compose(
```

Comment:
You could generally look at `withStateHandlers` for this logic, as it allows to define state and handlers in the same `withStateHandlers` HOC. And finally have keypress in a subsequent withHandlers. That should reduce the number of needed HOCs to only two.